src/App.tsx

```diff
@@ -1,61 +1,114 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const usersFromServer = [
+  { id: 1, name: 'Leanne Graham', email: 'Sincere@april.biz' },
+  { id: 2, name: 'Patricia Lebsack', email: 'Julianne.OConner@kory.org' },
+];
+
+const initialTodos = [
+  { id: 1, title: 'delectus aut autem', userId: 1, completed: true },
+  {
+    id: 2,
+    title: 'quis ut nam facilis et officia qui',
+    userId: 2,
+    completed: false,
+  },
+];
 
 export const App = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState<number | null>(null);
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+    setTitleError('');
+  };
+
+  const handleUserChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedUserId = parseInt(e.target.value, 10);
+
+    setUserId(selectedUserId);
+    setUserError(selectedUserId === 0 ? 'Please choose a user' : '');
+  };
+
+  const handleAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setTitleError('Please enter a title');
+    }
+
+    if (!userId) {
+      setUserError('Please choose a user');
+    }
+
+    if (!title.trim() || !userId) {
```

Comment:
Similar to the previous comment, the condition `if (!title.trim() || !userId)` should be adjusted to specifically check for `null` or `undefined` for `userId` to avoid treating `0` as falsy.